### PR TITLE
sunxi: OPP table, cpu voltage control and thermal zone for Orange Pi PC

### PR DESCRIPTION
**Pull request by uzer00**
Temperature info:
 cat /sys/class/thermal/thermal_zone0/temp
Scaling governors info (performance or powersave):
 cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
 echo "powersave" > /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor

Warning! For hard working system active cooling must be

Tests resulte with stress [menuconfig>tools>stress]:
Temp after test with passive cooling (aluminum heat sink) about 78-82C
Idle temp about 30-36C

>\#!/bin/sh
\#Test script
for i in 480000 816000 1008000 1296000 1440000 ; do
 echo "cur temp: \`cat /sys/class/thermal/thermal_zone0/temp\`"
 echo "set freq to: " $i
 echo $i >/sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
 stress -t 300 --cpu 4 --vm 2 --vm-bytes 512MB --io 4
 echo "temp after test: \`cat /sys/class/thermal/thermal_zone0/temp\`"
 sleep 60
done

Signed-off-by: Vitalij Alshevsky <v_alshevsky@tut.by>

### CODE REVIEW

**Comment by Menion2k** [comment]
Already proposed in PR #825

**Comment by hauke** [comment]
Are these the patches that are currently being proposed for upstream here: https://lkml.org/lkml/2018/2/5/900

**Comment by uzer00** [comment]
Yes, the patch based on Icenowy Zheng patches with some setups. I have only Orange Pi PC board for tests.

**Comment by diizzyy** [comment]
With the pending including of Linux kernel 4.19 it would perhaps be a more streamlined process bumping sunxi to 4.19 as soon as possible?

**Comment by diizzyy** [comment]
@uzer00 
I think we can close this and focus on 4.19 instead

**Comment by uzer00** [comment]
@diizzyy 
Tested and work fine

**Comment by ynezz** [comment]
Please see https://github.com/openwrt/openwrt/pull/1236#issuecomment-475022818, maybe you could unite the effort? Seems like you're working towards similar objective.

File: target/linux/sunxi/Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ FEATURES:=fpu usb ext4 display rtc squashfs
 SUBTARGETS:=cortexa8 cortexa7 cortexa53
 MAINTAINER:=Zoltan HERPAI <wigyori@uid0.hu>
 
-KERNEL_PATCHVER:=4.14
+KERNEL_PATCHVER:=4.19
```
Comment (ynezz) [inline target/linux/sunxi/Makefile]: Please don't do this, it's probably going to be changed for all targets at once 19.03 is branched.

**Comment by ynezz** [comment]
Please re-open the PR once you find a time (or interest) to work on this.

**Pull request closed without merging**